pkg/kubectl/cmd/get/customcolumn_test.go

```diff
@@ -373,3 +373,168 @@ bar    bar                bar
 		t.Errorf("\nexpected:\n'%s'\nsaw\n'%s'\n", expectedOutput, buffer.String())
 	}
 }
+
+func TestSliceColumnPrint(t *testing.T) {
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "fake-name",
+			Namespace: "fake-namespace",
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name: "fake0",
+				},
+				{
+					Name: "fake1",
+				},
+				{
+					Name: "fake2",
+				},
+				{
+					Name: "fake3",
+				},
+			},
+		},
+	}
+
+	tests := []struct {
+		name           string
+		spec           string
+		expectedOutput string
+	}{
+		{
+			name: "containers[0]",
+			spec: "CONTAINER:.spec.containers[0].name",
+			expectedOutput: `CONTAINER
+fake0
+`,
+		},
+		{
+			name: "containers[3]",
+			spec: "CONTAINER:.spec.containers[3].name",
+			expectedOutput: `CONTAINER
+fake3
+`,
+		},
+		// 		{
+		// 			name: "containers[-1], it equals containers[4]",
+		// 			spec: "CONTAINER:.spec.containers[-1].name",
+		// 			expectedOutput: `CONTAINER
```

Comment:
Did you mean to remove this?

Reply:
No, I commented them because it could trigger panic during test. Patch https://github.com/kubernetes/kubernetes/pull/72952 fix the panic.  Now that path has been merged, I will uncomment them right now.